test/jdk/javax/swing/JFrame/DefaultCloseOperation.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 1998, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.FlowLayout;
+import java.awt.Window;
+import java.awt.event.ItemEvent;
+import java.awt.event.WindowEvent;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+/*
+ * @test
+ * @summary test for defaultCloseOperation property for Swing JFrame and JDialog
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual DefaultCloseOperation
+ */
+
+public class DefaultCloseOperation extends JPanel {
+    private static final String INSTRUCTIONS = """
+        This is a manual test (requires user interaction) which tests the
```

Comment:
I guess the instructions should be amended so that it is clear what is required from the tester.

“To run this test, do the following steps…” could be replaced with “Do (Perform) the following steps…”.

jtreg automatically handles exceptions, it will fail there's an unhandled exception in the test.

Reply:
Cleaned up.